crates/astria-sequencer/src/accounts/state_ext.rs

```diff
@@ -69,8 +74,47 @@ fn nonce_storage_key<T: AddressBytes>(address: T) -> String {
 pub(crate) trait StateReadExt: StateRead + crate::assets::StateReadExt {
     #[instrument(skip_all)]
     async fn get_account_balances(&self, address: Address) -> Result<Vec<AssetBalance>> {
+        let ibc_balances = self
+            .get_account_balances_ibc(address)
+            .await
+            .context("failed to grab ibc balances for account")?;
+        let mut balances: Vec<AssetBalance> = Vec::with_capacity(ibc_balances.len());
+
+        for (asset, balance) in ibc_balances {
+            let native_asset = self
+                .get_native_asset()
+                .await
+                .context("failed to read native asset from state")?;
+            if asset == native_asset.to_ibc_prefixed() {
+                balances.push(AssetBalance {
+                    denom: native_asset.into(),
+                    balance,
+                });
+                continue;
+            }
+
+            let denom = self
+                .map_ibc_to_trace_prefixed_asset(asset)
+                .await
+                .context("failed to get ibc asset denom")?
+                .context("asset denom not found when user has balance of it; this is a bug")?
+                .into();
+            balances.push(AssetBalance {
+                denom,
+                balance,
+            });
+        }
+
+        Ok(balances)
+    }
+
+    #[instrument(skip_all)]
+    async fn get_account_balances_ibc<T: AddressBytes>(
```

Comment:
```suggestion
    async fn get_account_balances<T: AddressBytes>(
```

the native asset can still be "ibc-prefixed" (sort of confusing but it is what it is lol) so prefer to not specific `_ibc` here